### PR TITLE
[stable29] fix(unifiedSearch): Prevent broken avatars for federated users in peo…

### PR DESCRIPTION
**Pull request by nfebe**
| Before | After |
| --------- | ------ |
|Screenshot from 2024-04-22 16-08-33 | Screenshot from 2024-04-22 20-40-10 |

Resolves : https://github.com/nextcloud/server/issues/44319

### CODE REVIEW

File: core/src/components/UnifiedSearch/SearchableList.vue [hunk 1]
```diff
@@ -46,7 +46,8 @@
 						:wide="true"
 						@click="itemSelected(element)">
 						<template #icon>
-							<NcAvatar :user="element.user" :show-user-status="false" :hide-favorite="false" />
+							<NcAvatar v-if="element.isUser" :user="element.user" :show-user-status="false" :hide-favorite="false" />
```
Comment (Pytal) [inline core/src/components/UnifiedSearch/SearchableList.vue:49]:
```suggestion
							<NcAvatar v-if="element.isUser" :user="element.user" :show-user-status="false" />
```
Don't think this prop exists 🙈

File: core/src/components/UnifiedSearch/SearchableList.vue [hunk 2]
```diff
@@ -46,7 +46,8 @@
...
+							<NcAvatar v-else :is-no-user="true" :display-name="element.displayName" :show-user-status="false" :hide-favorite="false" />
```
Comment (Pytal) [inline core/src/components/UnifiedSearch/SearchableList.vue:50]:
```suggestion
							<NcAvatar v-else :is-no-user="true" :display-name="element.displayName" :show-user-status="false" />
```
Same here

**Review by Pytal — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**